internal/mesh/internal/mappers/sidecarproxymapper/service_endpoints_mapper.go

```diff
@@ -68,13 +58,31 @@ func (m *Mapper) MapServiceEndpointsToProxyStateTemplate(_ context.Context, _ co
 				continue
 			}
 			serviceRef := resource.Reference(serviceID, "")
-			if destination, ok := m.cache.ReadDestination(serviceRef, portName); ok {
+			if destination, ok := m.destinationsCache.ReadDestination(serviceRef, portName); ok {
 				for refKey := range destination.SourceProxies {
 					result = append(result, controller.Request{ID: refKey.ToID()})
 				}
 			}
 		}
 	}
 
+	// todo (ishustava): this is a stub for now until we implement implicit destinations.
+	// For tproxy, we generate requests for all proxy states in the cluster.
+	// This will generate duplicate events for proxies already added above,
+	// however, we expect that the controller runtime will de-dup for us.
+	rsp, err := rt.Client.List(ctx, &pbresource.ListRequest{
```

Comment:
>It feels like this could mask bugs in the above logic. Maybe it's worth adding a temporary flag to Mapper that we only use in tests to disable this?

I'm not sure I'm following. What kind of bugs you had in mind? I'm also not sure why we'd want to disable it in tests - we want this behavior for tests also.

>As a separate comment. What do you think about adding tests for this?

I didn't add unit tests for this specific function because it's a workaround until we have implicit destinations. This is tested though in the controller test that exercises all functionality with caches and mappers so I felt that this test coverage is sufficient for a workaround like this.